### PR TITLE
Fixed contract closed broken ui

### PR DESCRIPTION
**Pull request by sandeep-deriv**

### CODE REVIEW

File: static/css/_panel.scss [hunk 1]
```diff
@@ -67,6 +67,9 @@ $disabled-color: #F2F2F2;
         margin: 0 8.68% 0 8.68%;
         vertical-align: top;
         max-width: 120px;
+        @media (min-width: 1800px) {
+            margin: 0 8.60% 0 8.60%;
```
Comment (mehrzad-fs) [inline static/css/_panel.scss:71]:
```suggestion
            margin: 0 8.60%;
```

Comment (sandeep-deriv) [reply static/css/_panel.scss:71]: made suggested changes.

**Review by mehrzad-fs — APPROVED** [review]

File: src/botPage/view/TradeInfoPanel/AnimateTrade.js [hunk 2]
```diff
@@ -19,6 +20,7 @@ const resetAnimation = () => {
 };
 
 const activateStage = index => {
+    console.log(index, 'activateStage');
```
Comment (sara-fs) [inline src/botPage/view/TradeInfoPanel/AnimateTrade.js]: should we keep it here?

File: src/botPage/view/TradeInfoPanel/AnimateTrade.js [hunk 3]
```diff
@@ -27,66 +29,90 @@ const activateStage = index => {
...
         resetAnimation();
         setIndicatorMessage(INDICATOR_MESSAGES.not_running);
     };
 
-    const animateStage = contract_status => {
-        if (contract_status.id === 'contract.purchase_sent') {
+    const [contract_status, setContractStatus] = React.useState(CONTRACT_STATUS.not_running);
+
+    const animateStage = contract => {
+        if (contract.id === 'contract.purchase_sent') {
```
Comment (sara-fs) [inline src/botPage/view/TradeInfoPanel/AnimateTrade.js:37]: why should we this in the quotation marks? 'contract.purchase_sent'

File: src/botPage/view/TradeInfoPanel/AnimateTrade.js [hunk 4]
```diff
@@ -27,66 +29,90 @@ const activateStage = index => {
...
             $('.line').addClass('active');
             activateStage(1);
-            setBuyId(contract_status.data);
-        } else if (contract_status.id === 'contract.sold') {
+            setContractStatus(CONTRACT_STATUS.buy_succeeded);
+            // circle at 1: active
+            // bottom tooltip at 1: active
+
+            setBuyId(contract.data);
+        } else if (contract.id === 'contract.sold') {
```
Comment (sara-fs) [inline src/botPage/view/TradeInfoPanel/AnimateTrade.js:49]: you can add return to the If statements and remove `else` for better performance

File: src/botPage/view/TradeInfoPanel/AnimateTrade.js [hunk 5]
```diff
@@ -27,66 +29,90 @@ const activateStage = index => {
...
+            // circle at 2: actie
+            // bottom tooltip at 2: active
+            setSellId(contract.data);
         }
+        activateStage(contract.id);
+    };
 
-        activateStage(contract_status.id);
+    const onStop = () => {
+        console.log('onStop is called');
```
Comment (sara-fs) [inline src/botPage/view/TradeInfoPanel/AnimateTrade.js]: should we keep it here?

File: static/css/deriv/index.scss [hunk 6]
```diff
@@ -106,4 +106,172 @@ a:hover {
...
+        transform: translate(-50%, -50%);
+        height: 10px;
+        width: 10px;
+        border: 10px solid #fef1cf;
+        border-top-color: transparent;
+        border-left-color: transparent;
+        border-right-color: transparent;
+    }
+
+    // &.top {
```
Comment (sara-fs) [inline static/css/deriv/index.scss]: can we remove extra codes?

File: static/css/deriv/index.scss [hunk 7]
```diff
@@ -106,4 +106,172 @@ a:hover {
...
+    // }
+
+    p {
+        font-size: 11px;
+        text-align: center;
+        margin-top: -0.91em;
+        padding: 0.5em;
+        border-radius: 0.5em;
+        background-color: #fef1cf;
+        width: 120px;
```
Comment (sara-fs) [inline static/css/deriv/index.scss]: why sometimes `em` sometimes `px`?

File: static/css/deriv/index.scss [hunk 8]
```diff
@@ -106,4 +106,172 @@ a:hover {
...
+    }
+
+    &.running {
+        p {
+            color: #468847;
+            background-color: #DFF0D8;
```
Comment (sara-fs) [inline static/css/deriv/index.scss]: do we have any constants for these colors?

File: static/css/deriv/index.scss [hunk 9]
```diff
@@ -106,4 +106,172 @@ a:hover {
...
+            border-color: #D6E9C6;
+        }
+    }
+
+    &.active {
+        visibility: visible;
+        animation: slide-tooltip 0.8s 1;
```
Comment (sara-fs) [inline static/css/deriv/index.scss]: where is the` slide-tooltip `defined?

File: src/botPage/view/TradeInfoPanel/AnimateTrade.js (hunk 3 above)
Comment (sandeep-deriv) [reply src/botPage/view/TradeInfoPanel/AnimateTrade.js:37]: Screenshot 2022-01-28 at 10 19 45 AM


this is coming from back-end in this way.

File: src/botPage/view/TradeInfoPanel/AnimateTrade.js (hunk 4 above)
Comment (sandeep-deriv) [reply src/botPage/view/TradeInfoPanel/AnimateTrade.js:49]: done

File: static/css/deriv/index.scss (hunk 9 above)
Comment (sandeep-deriv) [reply static/css/deriv/index.scss]: it is inside the _panel.scss
I have moved the related code to _panel.scss

File: static/css/deriv/index.scss (hunk 8 above)
Comment (sandeep-deriv) [reply static/css/deriv/index.scss]: Only for these 3 color codes we don't have constants.
The occurrence of these colors is only once so I kept it as it is.
But later we can unify all the color codes and constants in a single file.

File: static/css/deriv/index.scss (hunk 6 above)
Comment (sandeep-deriv) [reply static/css/deriv/index.scss]: done

File: src/botPage/view/TradeInfoPanel/AnimateTrade.js (hunk 5 above)
Comment (sandeep-deriv) [reply src/botPage/view/TradeInfoPanel/AnimateTrade.js]: Removed...

**Review by sara-fs — APPROVED** [review]

File: src/botPage/view/TradeInfoPanel/AnimateTrade.js [hunk 10]
```diff
@@ -12,81 +13,71 @@ const INDICATOR_MESSAGES = {
...
+    attempting_to_buy: 'attempting_to_buy',
+    buy_succeeded: 'buy_succeeded',
+    contract_closed: 'contract_closed',
 };
 
 const AnimateTrade = () => {
     const [indicator_message, setIndicatorMessage] = React.useState(INDICATOR_MESSAGES.not_running);
     const [buy_price, setBuyPrice] = React.useState();
     const [buy_id, setBuyId] = React.useState();
     const [sell_id, setSellId] = React.useState();
```
Comment (mehrzad-fs) [inline src/botPage/view/TradeInfoPanel/AnimateTrade.js:27]: It's better to have an initial state for when we define useState

File: src/botPage/view/TradeInfoPanel/AnimateTrade.js [hunk 11]
```diff
@@ -12,81 +13,71 @@ const INDICATOR_MESSAGES = {
...
     const resetSummary = () => {
-        resetAnimation();
         setIndicatorMessage(INDICATOR_MESSAGES.not_running);
     };
 
-    const animateStage = contract_status => {
-        if (contract_status.id === 'contract.purchase_sent') {
-            resetAnimation();
-            activateStage(0);
+    const [contract_status, setContractStatus] = React.useState(CONTRACT_STATUS.not_running);
```
Comment (mehrzad-fs) [inline src/botPage/view/TradeInfoPanel/AnimateTrade.js:34]: please move this line before isMounted

File: src/botPage/view/TradeInfoPanel/AnimateTrade.js [hunk 12]
```diff
@@ -12,81 +13,71 @@ const INDICATOR_MESSAGES = {
...
+            return;
+        }
+
+        if (contract.id === 'contract.sold') {
+            setContractStatus(CONTRACT_STATUS.contract_closed);
+            setSellId(contract.data);
+        }
     };
 
     React.useEffect(() => {
```
Comment (mehrzad-fs) [inline src/botPage/view/TradeInfoPanel/AnimateTrade.js:60]: it's better if we define  React.useEffect as the first method after defining varialbes

File: src/botPage/view/TradeInfoPanel/AnimateTrade.js [hunk 13]
```diff
@@ -12,81 +13,71 @@ const INDICATOR_MESSAGES = {
...
         globalObserver.register('bot.running', () => {
-            $('.stage-tooltip.top:eq(0)').addClass('running');
             if (isMounted()) setIndicatorMessage(INDICATOR_MESSAGES.running);
         });
         globalObserver.register('bot.stop', () => {
-            $('.stage-tooltip.top:eq(0)').removeClass('running');
             if (isMounted()) setIndicatorMessage(INDICATOR_MESSAGES.stopped);
         });
 
         $('#stopButton').on('click', () => {
```
Comment (mehrzad-fs) [inline src/botPage/view/TradeInfoPanel/AnimateTrade.js:70]: is it possible to add `onClick` method for the `stopButton` and remove JQuery?

File: src/botPage/view/TradeInfoPanel/AnimateTrade.js [hunk 14]
```diff
@@ -12,81 +13,71 @@ const INDICATOR_MESSAGES = {
...
-            $('.stage-tooltip.top:eq(0)').removeClass('running');
             setIndicatorMessage(
                 globalObserver.getState('isRunning') ? INDICATOR_MESSAGES.stopping : INDICATOR_MESSAGES.stopped
             );
         });
+
         $('#runButton').on('click', () => {
```
Comment (mehrzad-fs) [inline src/botPage/view/TradeInfoPanel/AnimateTrade.js:76]: Same here

File: src/botPage/view/TradeInfoPanel/AnimateTrade.js [hunk 15]
```diff
@@ -95,57 +86,83 @@ const AnimateTrade = () => {
...
+
+const Stage = ({ title, classes: { circle, tooltip }, tooltip_text }) => (
+    <div className="trade-animator__stage">
+        <p className="title">{title}</p>
+        <span className={circle} />
+        <div className={tooltip}>
+            <p>{tooltip_text}</p>
+        </div>
+    </div>
+);
```
Comment (mehrzad-fs) [inline src/botPage/view/TradeInfoPanel/AnimateTrade.js:168]: Please define the stage component in a separate file or before `AnimiatedTrade` component

File: src/botPage/view/TradeInfoPanel/AnimateTrade.js (hunk 11 above)
Comment (sandeep-deriv) [reply src/botPage/view/TradeInfoPanel/AnimateTrade.js:34]: Done

**Pull request merged**